common/eos/src/eos_enclave_state.rs

```diff
@@ -45,13 +45,18 @@ impl EosEnclaveState {
         include_account_name: bool,
     ) -> Result<Self> {
         info!("✔ Getting EOS enclave state...");
+        #[cfg(not(feature = "spring_1-0"))]
         let incremerkles = Incremerkles::get_from_db(eos_db_utils)?;
-        let eos_last_seen_block_num = incremerkles.latest_block_num();
-        let eos_last_seen_block_id = incremerkles.latest_block_id()?.to_string();
 
         Ok(EosEnclaveState {
-            eos_last_seen_block_id,
-            eos_last_seen_block_num,
+            #[cfg(feature = "spring_1-0")]
+            eos_last_seen_block_id: eos_db_utils.get_eos_last_seen_block_id_from_db()?.to_string(),
+            #[cfg(not(feature = "spring_1-0"))]
+            eos_last_seen_block_id: incremerkles.latest_block_id()?.to_string(),
+            #[cfg(feature = "spring_1-0")]
+            eos_last_seen_block_num: eos_db_utils.get_latest_eos_block_number()?,
+            #[cfg(not(feature = "spring_1-0"))]
+            eos_last_seen_block_num: incremerkles.latest_block_num(),
```

Comment:
Unfortunately it's not possible to feature-gate a block in a struct constructor. I might make the whole struct conditional on the feature, but I chose to avoid that where possible, to keep changes as small as possible.
One exception would be something like [this](https://github.com/pnetwork-association/ptokens-core/pull/36/files#diff-8ac1c1f33eba56bc8cfd6f7eaaaa3e69846a03a1ca1bde4ac3dd7e60af8d3aabR171-R174):
```diff
+#[cfg(feature = "spring_1-0")]
+pub fn validate_block_header_signature<D: DatabaseInterface>(state: EosState<D>) -> Result<EosState<D>> {
+    Ok(state)
+}
+
+#[cfg(not(feature = "spring_1-0"))]
```

which was necessary not to impact too heavily either on the function, or on re-exports.

Reply:
I was expecting that, no worries then